### ISSUE TITLE
DigitalOcean setup with env variables is not working anymore

### ISSUE THREAD

**Issue by theoo** [rclone/rclone#8067]
#### What is the problem you are having with rclone?
Setting up DigitalOcean (at least) doesn't work using environment variables anymore since v.1.68.0

Note that I have no issue when the exact same parameters are set in the config file (`~/.config/rclone/rclone.conf`). 

#### What is your rclone version (output from `rclone version`)

```
# rclone --version
rclone v1.68.0
- os/version: debian 12.4 (64 bit)
- os/kernel: 6.1.0-22-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.23.1
- go/linking: static
- go/tags: none
```

and

```
» ./rclone --version
rclone v1.68.0
- os/version: arch (64 bit)
- os/kernel: 6.10.7-arch1-1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.23.1
- go/linking: static
- go/tags: none

```

Note that the same setup **works** fine with the version **v1.67.0**.

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Linux amd64, debian, ubuntu and arch

#### Which cloud storage system are you using? (e.g. Google Drive)

DigitalOcean

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

Any command that access the remove storage. In my use case: `ls` and `sync`.

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
# rclone ls do:EDIT-MY-BUCKET -vv
2024/09/11 13:05:00 DEBUG : Setting --s3-provider "DigitalOcean" from environment variable RCLONE_S3_PROVIDER="DigitalOcean"
2024/09/11 13:05:00 DEBUG : Setting --s3-access-key-id "EDIT" from environment variable RCLONE_S3_ACCESS_KEY_ID="EDIT"
2024/09/11 13:05:00 DEBUG : Setting --s3-secret-access-key "EDIT" from environment variable RCLONE_S3_SECRET_ACCESS_KEY="EDIT"
2024/09/11 13:05:00 DEBUG : Setting --s3-endpoint "fra1.digitaloceanspaces.com" from environment variable RCLONE_S3_ENDPOINT="fra1.digitaloceanspaces.com"
2024/09/11 13:05:00 DEBUG : Setting --s3-acl "public-read" from environment variable RCLONE_S3_ACL="public-read"
2024/09/11 13:05:00 DEBUG : rclone: Version "v1.68.0" starting with parameters ["rclone" "ls" "do:EDIT-MY-BUCKET" "-vv"]
2024/09/11 13:05:00 DEBUG : Creating backend with remote "do:EDIT-MY-BUCKET"
2024/09/11 13:05:00 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2024/09/11 13:05:00 DEBUG : do: detected overridden config - adding "{XHDCm}" suffix to name
2024/09/11 13:05:00 DEBUG : fs cache: renaming cache item "do:EDIT-MY-BUCKET" to be canonical "do{XHDCm}:EDIT-MY-BUCKET"
SDK 2024/09/11 13:05:00 WARN falling back to IMDSv1: operation error ec2imds: getToken, http response error StatusCode: 405, request to EC2 IMDS failed
2024/09/11 13:05:00 DEBUG : 4 go routines active
2024/09/11 13:05:00 NOTICE: Failed to ls: directory not found
```

My config file: 
```
[do]
type = s3
env_auth = true
```

Environment variables set:
```
export RCLONE_S3_ACCESS_KEY_ID=EDIT
export RCLONE_S3_SECRET_ACCESS_KEY=EDIT
export RCLONE_S3_ENDPOINT=EDIT
export RCLONE_S3_PROVIDER=DigitalOcean
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by Animosity022** [comment]
See https://github.com/rclone/rclone/issues/8063

as it's a duplicate of that one.

**Comment by ncw** [comment]
I don't think this is a duplicate of #8063 - I think it is most likely to do with the SDK v2 re-organisation

I suspect this is the problem here

> env_auth = true

You are telling rclone to look for AWS authentication (eg `~/.aws/config` the AWS environment variables, eg 

- Access Key ID: `AWS_ACCESS_KEY_ID` or `AWS_ACCESS_KEY`
- Secret Access Key: `AWS_SECRET_ACCESS_KEY` or `AWS_SECRET_KEY`
- Session Token: `AWS_SESSION_TOKEN` (optional)

You don't need to do that because you are using rclone config variables, eg `RCLONE_S3_ACCESS_KEY_ID=EDIT`

So if you remove that line from your config I think it will work.

If that does work it means that somehow we've introduced a backwards incompatibility into rclone which needs looking at.

**Comment by theoo** [comment]
Thanks for your feedback @ncw and reopening the ticket without any animosity. :drum: Pun intended :bell: 

Using the follwing install script provided on [the download page](https://rclone.org/downloads/)

```bash
curl https://rclone.org/install.sh | sudo bash
```

simply caused our pipeline to stop working. rclone 1.68.0 has a breaking change and I just rolled back to 1.67.0 for now.

This being said, I quickly tried your recommandation. Setting `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` instead of `RCLONE_S3_ACCESS_KEY_ID` and `RCLONE_S3_SECRET_ACCESS_KEY` **alongside** with `RCLONE_S3_ENDPOINT` and `RCLONE_S3_PROVIDER` worked on my computer, so this should work as a work-around for people willing to setup _rclone 1.68.0_ with env vars.

I can imagine that other users are in the same situation. CI/CD variables in GitLab or _secrets_ in k8s are often used to set environment variable in pipeline/container. Deployments is probably one of the important use-cases for _rclone_.

`env_auth = true` is indeed the default setting when calling it without config file, this value has not been set explicity.

**Comment by Animosity022** [comment]
> Thanks for your feedback @ncw and reopening the ticket without any animosity. 🥁 Pun intended 🔔
> 
> Using the follwing install script provided on [the download page](https://rclone.org/downloads/)
> 
> ```shell
> curl https://rclone.org/install.sh | sudo bash
> ```
> 
> simply caused our pipeline to stop working. rclone 1.68.0 has a breaking change and I just rolled back to 1.67.0 for now.
> 
> This being said, I quickly tried your recommandation. Setting `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` instead of `RCLONE_S3_ACCESS_KEY_ID` and `RCLONE_S3_SECRET_ACCESS_KEY` **alongside** with `RCLONE_S3_ENDPOINT` and `RCLONE_S3_PROVIDER` worked on my computer, so this should work as a work-around for people willing to setup _rclone 1.68.0_ with env vars.
> 
> I can imagine that other users are in the same situation. CI/CD variables in GitLab or _secrets_ in k8s are often used to set environment variable in pipeline/container. Deployments is probably one of the important use-cases for _rclone_.
> 
> `env_auth = true` is indeed the default setting when calling it without config file, this value has not been set explicity.

haha - that made me laugh - I do get it wrong here/there so Nick always makes sure to catch those ones :)

**Comment by ncw** [comment]
I've fixed that little regression now @theoo - can you try this with your original config?

[v1.69.0-beta.8301.21f9d3a30.fix-8067-s3-config](https://beta.rclone.org/branch/fix-8067-s3-config/v1.69.0-beta.8301.21f9d3a30.fix-8067-s3-config/) on branch [fix-8067-s3-config](https://github.com/rclone/rclone/tree/fix-8067-s3-config) (uploaded in 15-30 mins)

It is explicitly documented the old behavior, so we shouldn't break it!

Thanks

**Comment by theoo** [comment]
Thanks for your patience @ncw . I ran the test with the above provided fix and it works on my computer and pipelines with the following variables set:

#### RCLONE Env Vars, env_auth = [false | true]

`RCLONE_S3_ENDPOINT`
`RCLONE_S3_ACCESS_KEY_ID`
`RCLONE_S3_SECRET_ACCESS_KEY`

#### AWS Env Vars, env_auth = true

`RCLONE_S3_ENDPOINT`
`AWS_ACCESS_KEY_ID`
`AWS_SECRET_ACCESS_KEY`

So I can confirm that this patch fixes the regression from my side too.
Many thanks for your quick response and fix.

**Comment by ncw** [comment]
Thanks for testing @theoo 

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.69 and v1.68.1
